main/src/main/scala/sbt/Main.scala

```diff
@@ -92,7 +92,7 @@ final class xMain extends xsbti.AppMain {
   private def endsWithBoot(state: State) = state.remainingCommands.lastOption exists (_.commandLine == BootCommand)
 
   private def notifyUsersAboutShell(state: State) =
-    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, Server) && !endsWithBoot(state)) {
+    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, OldShell) && !endsWithBoot(state)) {
```

Comment:
I should've noticed the trailing commits in https://github.com/sbt/sbt/pull/3077